internal/db/migrations/005_reservations.sql

```diff
@@ -1,8 +1,9 @@
 CREATE TABLE reservations
 (
   id BIGINT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+  provider INTEGER NOT NULL CHECK (valid_provider(provider)),
```

Comment:
can you also add source_id? I believe we will need it, so once we adjust the migration already...